File: api/src/main/java/br/cefet/aida/model/Jogo.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package br.cefet.aida.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 *
 * @author dougl
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Jogo {
    private int idJogo;
    private int idTorneio;
    private int idEquipeA;
    private int rodada;
    private Equipe equipeA;
    private int idEquipeB;
    private Equipe equipeB;
    private int idVencedor;
    private Equipe equipeVencedor;
}

File: site/src/app/model/jogo.ts
import { Equipe } from "./equipe";

export class Jogo {
    idJogo: number;
    rodada: number;
    idTorneio: number;
    idEquipeA: number;
    equipeA: Equipe;
    idEquipeB: number;
    equipeB: Equipe;
    idVencedor: number;
    equipeVencedor: Equipe;

    constructor() {
        this.idJogo = 0;
        this.idTorneio = 0;
        this.idEquipeA = 0;
        this.rodada = 0;
        this.equipeA = new Equipe();
        this.idEquipeB = 0;
        this.equipeB = new Equipe;
        this.idVencedor = 0;
        this.equipeVencedor = new Equipe();
    }
}
